proto/internal/temporal/server/api/cli/v1/message.proto

```diff
@@ -54,7 +54,7 @@ message PendingActivityInfo {
     string activity_id = 1;
     temporal.api.common.v1.ActivityType activity_type = 2;
     temporal.api.enums.v1.PendingActivityState state = 3;
-    string heartbeat_details = 4;
+    repeated string heartbeat_details = 4;
```

Comment:
I'm actually confused why heartbeat details are modeled as string.  I think this should be blob like we model where user Payload(HHeader, Data) is passed through the system.  @alexshtin can you confirm if this is something we missed and needs to be converted to Payload type.

Reply:
yeah, i can plumb the encoding and the payload directly through as well. good point on questioning the assumption on this being a string

Reply:
the recordheartbeat and getheartbeat activity functions do take in whole objects. the "header" is part of our internal de/serialization mechanism where we decide whether to write a raw byte array directly vs. convert the object to json. so I can understand why we wouldn't want to at least surface the header part out of the CLI.

This change breaks if we try to record a raw byte array though, but I think getting raw byte array details out of the CLI has been broken for a while

Reply:
moved bulk of this logic into SDK. Diff is smaller now.